workflow/controller/operator.go

```diff
@@ -2108,6 +2107,43 @@ func (woc *wfOperationCtx) onNodeComplete(node *wfv1.NodeStatus) {
 	)
 }
 
+// recordNodePhaseChangeEvents creates WorkflowNode Kubernetes events for each node
+// that has changes logged during this execution of the operator loop.
+func (woc *wfOperationCtx) recordNodePhaseChangeEvents(old *wfv1.Nodes, new *wfv1.Nodes) {
+	if !woc.controller.Config.NodeEvents.IsEnabled() {
+		return
+	}
+	oldNodes := *old
+	newNodes := *new
```

Comment:
I think you should to DeepCopy(), that way you reduce risk of bugs getting introduced